sequencer/src/block.rs

```diff
@@ -2,26 +2,101 @@ use crate::{
     l1_client::{L1Client, L1ClientOptions, L1Snapshot},
     L1BlockInfo, NMTRoot, NamespaceProofType, Transaction, TransactionNMT, VmId, MAX_NMT_DEPTH,
 };
-use ark_serialize::CanonicalSerialize;
+use ark_serialize::{
+    CanonicalDeserialize, CanonicalSerialize, Compress, Read, SerializationError, Valid, Validate,
+    Write,
+};
 use async_std::task::{block_on, sleep};
 use commit::{Commitment, Committable, RawCommitmentBuilder};
 use hotshot_query_service::availability::QueryablePayload;
 use hotshot_types::{
     data::VidCommitment,
     traits::block_contents::{vid_commitment, BlockHeader, BlockPayload},
 };
-use jf_primitives::merkle_tree::{
-    namespaced_merkle_tree::NamespacedMerkleTreeScheme, LookupResult, MerkleCommitment,
-    MerkleTreeScheme,
+use jf_primitives::{
+    errors::PrimitivesError,
+    merkle_tree::{
+        light_weight::LightWeightMerkleTree, namespaced_merkle_tree::NamespacedMerkleTreeScheme,
+        AppendableMerkleTreeScheme, DigestAlgorithm, Element, Index, LookupResult,
+        MerkleCommitment, MerkleTreeScheme,
+    },
 };
 use lazy_static::lazy_static;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use sha3::{Digest, Sha3_256};
 use std::{
     env,
     fmt::{Debug, Display},
     time::Duration,
 };
 use time::OffsetDateTime;
+use typenum::U3;
+
+/// Update the array length here
+#[derive(Default, Eq, PartialEq, Clone, Copy, Debug, Ord, PartialOrd, Hash)]
+pub struct Sha3Node(pub(crate) [u8; 32]);
```

Comment:
@jbearer @mrain is this the right hash function for the block merkle tree?

Reply:
SHA3 should be fine but it looks like Jellyfish already provides a SHA3 hash scheme: https://jellyfish.docs.espressosys.com/jf_primitives/merkle_tree/examples/struct.Sha3Digest.html. Not sure why it's in examples, though? @mrain?

In any case I don't think we should be implementing our own hash scheme here since it's not anything sequencer-specific. We should use the one from Jellyfish or update it if necessary.